core/commands/pin/remotepin.go

```diff
@@ -669,3 +673,16 @@ func getRemotePinServiceInfo(env cmds.Environment, name string) (url, key string
 	}
 	return service.Api.Endpoint, service.Api.Key, nil
 }
+
+func normalizeEndpoint(endpoint string) (string, error) {
+	uri, err := neturl.ParseRequestURI(endpoint)
+	if err != nil || !strings.HasPrefix(uri.Scheme, "http") {
+		return "", fmt.Errorf("service endpoint must be a valid HTTP URL")
+	}
+	// avoid //pins (https://github.com/ipfs/go-ipfs/issues/7826)
+	uri.Path = strings.TrimSuffix(uri.Path, "/")
```

Comment:
https://github.com/ipfs/go-ipfs/issues/7826 seems to be asking for path normalization. IIUC this could include any number of repeated `/`s in the uri and I'm not sure if there are any other normalization considerations.

Perhaps Go's `path` package's Clean function could be of use here, or maybe there's something more specific to URIs.

Reply:
I went with `path.Clean` and added more tests:  2ffdce500d5a788a19fbff1e1d4c8c5220bf3df2